### PR TITLE
Add debug log when encounters permission check failure

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
The main changes are:
1. Add some debug log when there's permission check failure, the format is as follows:
`Authorization failed, principals: $principals has no $permissionModes on the requested saved object: $savedObjects`
, here is an example:
`Authorization failed, principals: {"groups":[""],"users":["user_read_tenant"]} has no [read] permissions on the requested saved object: [{"id":"test1-3.0.0","type":"config","workspaces":["test1"],"permissions":{"library_write":{"groups":["admin"],"users":["test_user","user_rw_tenant"]}}}]`

2. Found that if the requested saved object doesn't exist, currently the `permissionControl.validate` function returns `{"success":true, "result":true}`, that's because when the saved object doesn't exist, `bulkGetSavedObjects` returns an object just like:
```
[
  {
    id: 'test1-3.0.1',
    type: 'config',
    error: {
      statusCode: 404,
      error: 'Not Found',
      message: 'Saved object [config/test1-3.0.1] not found'
    }
  }
]
```
, I think we should handle this case carefully and return the error to the caller:

```
{"success":false,"error":{"statusCode":404,"error":"Not Found","message":"Saved object [config/test1-3.0.1] not found"}}
``` 

## Screenshot

image

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/server/saved_objects/permission_control/client.ts [hunk 1]
```diff
@@ -86,27 +93,53 @@ export class SavedObjectsPermissionControl {
...
+    if (!hasAllPermission) {
+      const savedObjectsBasicInfo = savedObjectsGet.map((item) => {
+        return {
+          id: item.id,
+          type: item.type,
+          workspaces: item.workspaces,
+          permissions: item.permissions,
+        };
+      });
+      this.logger.debug(
```
Comment (ruanyl) [inline src/core/server/saved_objects/permission_control/client.ts:130]: We can use [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) which is easier to use

Comment (gaobinlong) [reply src/core/server/saved_objects/permission_control/client.ts:130]: I've changed that, thanks!

**Review by ruanyl — APPROVED** [review]

File: src/core/server/saved_objects/permission_control/client.ts [hunk 2]
```diff
@@ -85,27 +92,51 @@ export class SavedObjectsPermissionControl {
...
+        `Authorization failed, principals: ${JSON.stringify(
+          principals
+        )} has no [${permissionModes}] permissions on the requested saved object: ${JSON.stringify(
+          savedObjectsBasicInfo
+        )}`
+      );
```
Comment (Hailong-am) [inline src/core/server/saved_objects/permission_control/client.ts:135]: seems that we log all the saved objects in request here, i am not sure whether we should log the saved objects which user don't have permission or all of them?

Comment (gaobinlong) [reply src/core/server/saved_objects/permission_control/client.ts:140]: Only log the saved objects which user has no permission is better, I've change that, please help to take a look.

**Review by Hailong-am — APPROVED** [review]

**Pull request merged**